File: src/main/scala/com/chyuck/algorithms/scala/SelectionSort.scala
package com.chyuck.algorithms.scala

object SelectionSort {

  def sort(array: Array[Int]) : Array[Int] = {
    if (array.length <= 1) array
    else {

      for (i <- 0 until array.length) {

        var minValueIndex = i

        for (j <- i + 1 until array.length)
          if (array(j) < array(minValueIndex))
            minValueIndex = j

        val temp = array(i)
        array(i) = array(minValueIndex)
        array(minValueIndex) = temp
      }

      array
    }
  }

}

File: src/main/java/com/chyuck/algorithms/java/SelectionSort.java
package com.chyuck.algorithms.java;

public class SelectionSort {

    public static int[] sort(int[] array) throws IllegalArgumentException{
        if (array == null) throw new IllegalArgumentException("array is null");

        if (array.length <= 1) return array;

        for (int i = 0; i < array.length; i++) {

            int minValueIndex = i;

            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[minValueIndex])
                    minValueIndex = j;
            }

            int temp = array[i];
            array[i] = array[minValueIndex];
            array[minValueIndex] = temp;
        }

        return array;
    }
}
